src/Umbraco.Web.UI.Client/src/packages/user/user/workspace/user/components/user-workspace-avatar/user-workspace-avatar.element.ts

```diff
@@ -91,7 +114,7 @@ export class UmbUserAvatarElement extends UmbLitElement {
 						.name=${this._user.name}
 						.kind=${this._user.kind}
 						.imgUrls=${this._user.avatarUrls ?? []}></umb-user-avatar>
-					<input id="AvatarFileField" type="file" name="avatarFile" required hidden />
+					<input id="AvatarFileField" type="file" name="avatarFile" accept=${this._allowedFileTypes} required hidden />
```

Comment:
You can get the file size through the input change event. Is that enough to prevent the upload before sending it to the repo?

Something like this:

```ts	
#onAvatarFileChange(event: Event) {
  const target = event.target as HTMLInputElement;
  const file = target.files?.[0];
  const fileSize = file?.size;
  console.log('File size:', fileSize);
}```

Reply:
Oh, but do I need that? It looks like it passes the file on to the UserAvatarRepository, which then uses the UmbTemporaryFileManager, which validates the allowed file size.

Reply:
Ah, no problem then.

I have misunderstood this: 
> If the developer fails to inform us of the intended max file size (also known as the max request length), we have no way of using this to validate the file size before attempting an upload.

Reply:
Ohhh, I see! Yeah, that's on the server side, i.e., they have to tell us what their web server is configured to, because we cannot automagically guess when the web server blows up in an error. We have a configuration value for that here:

```json
{
  "Umbraco": {
    "CMS": {
      "Runtime": {
        "MaxRequestLength": 1 <-- Tell us what your web server accepts (in kilobytes)
      }
    }
  }
}
```